### PR TITLE
Add monotonicity fixer

### PR DESCRIPTION
**Pull request by karlfloersch**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   },
   "dependencies": {
     "@eth-optimism/contracts": "^0.0.2-alpha.7",
+    "new-contracts": "npm:@eth-optimism/contracts@0.1.4",
```
Comment (tynes) [inline package.json:31]: You can't just do an upgrade?

File: src/batch-submitter/tx-batch-submitter.ts [hunk 2]
```diff
@@ -380,12 +436,38 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
...
-    if (this.autoFixBatchOptions.fixDelayedTimestampAndBlockNumberHardcoded) {
-      batch = await fixDelayedTimestampAndBlockNumberHardcoded(batch)
+    if (this.autoFixBatchOptions.fixMonotonicity) {
+      batch = await fixMonotonicity(batch)
     }
     return batch
   }
 
+  private async _getLastTimestampAndBlockNumber(): Promise<{ lastTimestamp: number, lastBlockNumber: number}> {
+    const addressManagerAddr = (await this._getRollupInfo()).addresses.addressResolver
```
Comment (tynes) [inline src/batch-submitter/tx-batch-submitter.ts:446]: This is going to be removed in the new geth, we should pull the address manager address from a config option

**Comment by tynes** [comment]
This generally looks good to me after reading through it and observing it submit batches from the PoV of a verifier

File: package.json (hunk 1 above)
Comment (karlfloersch) [reply package.json:31]: Got an error when upgrading and got frustrated trying to upgrade the batch submitter to the newer version. Definitely tech debt and gotta go back and upgrade everything to the latest version 😭

File: src/batch-submitter/tx-batch-submitter.ts (hunk 2 above)
Comment (karlfloersch) [reply src/batch-submitter/tx-batch-submitter.ts:446]: Ok sounds good

**Comment by karlfloersch** [comment]
Ok well that's good but I really badly want to rewrite the batch submitter and rip out all this 'self healing' stuff 😭

File: package.json (hunk 1 above)
Comment (tynes) [reply package.json:31]: I had a similar error and fixed it by deleting the `yarn.lock`. Does that fix it?

Comment (karlfloersch) [reply package.json:31]: Just tried and also downgraded to 1.4 and no luck 😕

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Will definitely cause a merge conflict with https://github.com/ethereum-optimism/batch-submitter/pull/32

**Pull request merged**